### ISSUE TITLE
TokenError for incomplete command PyodideConsole

### ISSUE THREAD

**Issue by leopsidom** [pyodide/pyodide#1765]
I'm getting a token error on a command that seems to be incomplete. Steps to reproduce:

```
 pyodide = await loadPyodide({ indexURL: "https://cdn.jsdelivr.net/pyodide/v0.18.0a1/full" })
 pyodide.runPython(`
import js
import __main__
from pyodide.console import PyodideConsole
js.pyconsole = PyodideConsole(__main__.__dict__)`)
```

And then run:

```
 fut = pyconsole.push("[x for x in")
```

I'm getting this error:

>  Uncaught PythonError: Traceback (most recent call last):
  File "/lib/python3.9/site-packages/_pyodide/console.py", line 421, in push
    result = self.runsource(source, self.filename)
  File "/lib/python3.9/site-packages/_pyodide/console.py", line 328, in runsource
    code = self._compile(source, filename, "single")
  File "/lib/python3.9/site-packages/_pyodide/console.py", line 142, in __call__
    return super().__call__(source, filename, symbol)  # type: ignore
  File "/lib/python3.9/codeop.py", line 178, in __call__
    return _maybe_compile(self.compiler, source, filename, symbol)
  File "/lib/python3.9/codeop.py", line 83, in _maybe_compile
    code = compiler(source, filename, symbol)
  File "/lib/python3.9/site-packages/_pyodide/console.py", line 100, in __call__
    if self.quiet_trailing_semicolon and should_quiet(source):
  File "/lib/python3.9/site-packages/_pyodide/_base.py", line 40, in should_quiet
    tokens = list(tokenize.generate_tokens(source_io.readline))
  File "/lib/python3.9/tokenize.py", line 521, in _tokenize
    raise TokenError("EOF in multi-line statement", (lnum, 0))
tokenize.TokenError: ('EOF in multi-line statement', (6, 0))

While comparing to `code.compile_command`, `code.compile_command` gives result as `incomplete`:

```
import code
c1 = code.compile_command("[x for x in")
type(c1)
<class 'NoneType'>
```

Wonder if this is this expected behavior of the PyodideConsole ?

**Comment by leopsidom** [comment]
Okay I think this CR: https://github.com/pyodide/pyodide/pull/1759/commits is fixing the issue. But it might not be included in 0.18.0a1 release yet.

**Comment by hoodmane** [comment]
Indeed, #1759 fixes this and is not included in 0.18.0a1. Thanks for these bug reports @leopsidom, we really appreciate the feedback!

**Comment by leopsidom** [comment]
Sure, thanks for proactively fixing the issue. Will the fix be included in the upcoming 0.18.0 release ? Also wonder if there will be another `a2`  release before the official one, considering it has been pretty close to the official release date ?

**Comment by hoodmane** [comment]
> Will the fix be included in the upcoming 0.18.0 release ?

Yeah definitely.

> wonder if there will be another a2 release

We weren't planning on it due to laziness but it could be a good idea to do an a2 on Tuesday August 3rd and push the 18.0 release to Tuesday August 10th. I don't have much experience making software releases so I don't have any traditions to draw on when deciding how best to approach this. It would be helpful if you would suggest it here: #1738.

**Comment by leopsidom** [comment]
Thanks for clarification. I have made a comment in the thread.
